lib/ansible/modules/network/cloudengine/ce_vxlan_tunnel.py

```diff
@@ -0,0 +1,936 @@
+            if not self.is_vni_peer_list_exist(nve_name, vni_id, peer_ip):
+                self.module.fail_json(msg='Error: The %s is not exist' % peer_ip)
+        config = self.get_current_config(vni_id, peer_ip_list)
+        if not config:
+            cfg_xml = CE_NC_DELETE_VNI_PEER_ADDRESS_IP_HEAD % (
+                nve_name, vni_id)
+            for peer_ip in peer_ip_list:
+                cfg_xml += CE_NC_DELETE_VNI_PEER_ADDRESS_IP_DELETE % peer_ip
+            cfg_xml += CE_NC_DELETE_VNI_PEER_ADDRESS_IP_END
+        else:
+            cfg_xml = CE_NC_DELETE_PEER_ADDRESS_IP_HEAD % (
+                nve_name, vni_id)
+            for peer_ip in peer_ip_list:
+                cfg_xml += CE_NC_DELETE_VNI_PEER_ADDRESS_IP_DELETE % peer_ip
+            cfg_xml += CE_NC_DELETE_PEER_ADDRESS_IP_END
+
+        recv_xml = set_nc_config(self.module, cfg_xml)
+        self.check_response(recv_xml, "DELETE_VNI_PEER_IP")
+        self.updates_cmd.append("interface %s" % nve_name)
+
+        for peer_ip in peer_ip_list:
+            cmd_output = "undo vni %s head-end peer-list %s" % (vni_id, peer_ip)
+            self.updates_cmd.append(cmd_output)
+
+        self.changed = True
+
+    def config_delete_vni_protocol_type(self, nve_name, vni_id, protocol_type):
+        """remove vni protocol type"""
+
+        if not self.is_vni_protocol_exist(nve_name, vni_id, protocol_type):
+            return
+
+        cfg_xml = CE_NC_DELETE_VNI_PROTOCOL % (nve_name, vni_id, protocol_type)
+        recv_xml = set_nc_config(self.module, cfg_xml)
+        self.check_response(recv_xml, "DELETE_VNI_PEER_PROTOCOL")
+        self.updates_cmd.append("interface %s" % nve_name)
+        self.updates_cmd.append(
+            "undo vni %s head-end peer-list protocol bgp " % vni_id)
+        self.changed = True
+
+    def check_params(self):
+        """Check all input params"""
+
+        # bridge_domain_id check
+        if self.bridge_domain_id:
+            if not self.bridge_domain_id.isdigit():
+                self.module.fail_json(
+                    msg='Error: The parameter of bridge domain id is invalid.')
+            if int(self.bridge_domain_id) > 16777215 or int(self.bridge_domain_id) < 1:
+                self.module.fail_json(
+                    msg='Error: The bridge domain id must be an integer between 1 and 16777215.')
+        # vni_id check
+        if self.vni_id:
+            if not self.vni_id.isdigit():
+                self.module.fail_json(
+                    msg='Error: The parameter of vni id is invalid.')
+            if int(self.vni_id) > 16000000 or int(self.vni_id) < 1:
+                self.module.fail_json(
+                    msg='Error: The vni id must be an integer between 1 and 16000000.')
+
+        # nve_name check
+        if self.nve_name:
+            if not self.check_nve_name():
+                self.module.fail_json(
+                    msg='Error: Error: Nve interface is invalid.')
```

Comment:
Perhaps include the name of the NVE interface in the error message?
Also `Nve` or `NVE`?